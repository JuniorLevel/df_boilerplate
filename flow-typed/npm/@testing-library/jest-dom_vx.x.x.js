// flow-typed signature: d55413c6ce1fdd571999428ccc34f041
// flow-typed version: <<STUB>>/@testing-library/jest-dom_v^6.5.0/flow_v0.245.2
// @flow

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@testing-library/jest-dom'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@testing-library/jest-dom' {
	declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@testing-library/jest-dom/dist' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/dist/jest-globals' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/dist/matchers-4fe91ec3' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/dist/matchers' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/dist/vitest' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/jest-globals' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/matchers' {
	declare module.exports: any;
}

declare module '@testing-library/jest-dom/vitest' {
	declare module.exports: any;
}

// Filename aliases
declare module '@testing-library/jest-dom/dist/index' {
	declare module.exports: $Exports<'@testing-library/jest-dom/dist'>;
}
declare module '@testing-library/jest-dom/dist/index.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/dist'>;
}
declare module '@testing-library/jest-dom/dist/jest-globals.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/dist/jest-globals'>;
}
declare module '@testing-library/jest-dom/dist/matchers-4fe91ec3.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/dist/matchers-4fe91ec3'>;
}
declare module '@testing-library/jest-dom/dist/matchers.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/dist/matchers'>;
}
declare module '@testing-library/jest-dom/dist/vitest.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/dist/vitest'>;
}
declare module '@testing-library/jest-dom/jest-globals.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/jest-globals'>;
}
declare module '@testing-library/jest-dom/matchers.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/matchers'>;
}
declare module '@testing-library/jest-dom/vitest.js' {
	declare module.exports: $Exports<'@testing-library/jest-dom/vitest'>;
}
