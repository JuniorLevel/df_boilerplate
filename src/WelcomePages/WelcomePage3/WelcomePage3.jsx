// @flow

import React from 'react';
import MediaQuery from 'react-responsive';
import { Welcome } from '../../lib/components/Welcome/Welcome';
import { WelcomeAuthButtons } from '../../lib/components/Welcome/WelcomeAuthButtons/WelcomeAuthButtons';
import { FourImagesBg } from '../../lib/components/Welcome/FourImagesBg/FourImagesBg';
import { WelcomeButton } from '../../lib/components/Welcome/WelcomeAuthButtons/WelcomeButton/WelcomeButton';
import { welcomeButtons } from './data';

interface IWelcomePage3Props {
	imagesUrls: { id: number, url: string }[];
}

export const WelcomePage3 = ({
	imagesUrls,
}: IWelcomePage3Props): React.Node => (
	<FourImagesBg imagesUrls={imagesUrls}>
		<Welcome buttons={welcomeButtons}>
			<MediaQuery minWidth={361}>
				<WelcomeAuthButtons>
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
				</WelcomeAuthButtons>
			</MediaQuery>
			<MediaQuery maxWidth={360}>
				<WelcomeAuthButtons>
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
				</WelcomeAuthButtons>
			</MediaQuery>
		</Welcome>
	</FourImagesBg>
);
