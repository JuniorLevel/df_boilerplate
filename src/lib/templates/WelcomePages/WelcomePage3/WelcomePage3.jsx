// @flow

import React from 'react';
import MediaQuery from 'react-responsive';
import { Welcome } from '../../../components/Welcome/Welcome';
import { WelcomeAuthButtons } from '../../../components/Welcome/WelcomeAuthButtons/WelcomeAuthButtons';
import { FourImagesBg } from '../../../components/Welcome/FourImagesBg/FourImagesBg';
import { buttons } from '../../../components/Welcome/buttons';
import { WelcomeButton } from '../../../components/Welcome/WelcomeAuthButtons/WelcomeButton/WelcomeButton';
import Image1 from '../../../assets/Image1.jpg';
import Image2 from '../../../assets/Image2.jpg';
import Image3 from '../../../assets/Image3.jpg';
import Image4 from '../../../assets/Image4.jpg';

const images = [
	{ id: 1, url: Image1 },
	{ id: 2, url: Image2 },
	{ id: 3, url: Image3 },
	{ id: 4, url: Image4 },
];

const WelcomePage3 = (): React.Node => (
	<FourImagesBg imagesUrls={images}>
		<Welcome buttons={buttons}>
			<MediaQuery minWidth={361}>
				<WelcomeAuthButtons>
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
				</WelcomeAuthButtons>
			</MediaQuery>
			<MediaQuery maxWidth={360}>
				<WelcomeAuthButtons>
					<WelcomeButton title="G" onClick={(): any => {}} />
					<WelcomeButton title="VK" onClick={(): any => {}} />
				</WelcomeAuthButtons>
			</MediaQuery>
		</Welcome>
	</FourImagesBg>
);

export default WelcomePage3;
