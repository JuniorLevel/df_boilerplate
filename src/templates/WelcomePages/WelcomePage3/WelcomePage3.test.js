import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import WelcomePage3 from './WelcomePage3';

describe('WelcomePage2', () => {
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<WelcomePage3 />
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});

	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<WelcomePage3 />
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});

	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<WelcomePage3 />
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
