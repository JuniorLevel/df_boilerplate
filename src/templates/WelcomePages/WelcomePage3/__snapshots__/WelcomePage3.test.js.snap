// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage2 matches the snapshot desktop 1`] = `
<div>
  <div
    class="FourImagesBgstyles__DivImageContainer-sc-1fpm6f5-0 ckwQRQ"
  >
    <img
      alt="landscape"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <img
      alt="space"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <img
      alt="landscape 3"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <img
      alt="landscape 4"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-dev-only-do-not-override-1uq9j6g"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-e8hzxi-0 iNYXqo"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              Create Account
            </span>
          </button>
        </div>
        <div
          class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-wrap-wrap ant-flex-justify-center ant-flex-gap-large"
          style="margin-bottom: 20px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomePage2 matches the snapshot laptop 1`] = `
<div>
  <div
    class="FourImagesBgstyles__DivImageContainer-sc-1fpm6f5-0 ckwQRQ"
  >
    <img
      alt="landscape"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <img
      alt="space"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <img
      alt="landscape 3"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <img
      alt="landscape 4"
      class="FourImagesBgstyles__Image-sc-1fpm6f5-1 vpGEF"
      src="[object Object]"
    />
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-dev-only-do-not-override-1uq9j6g"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-e8hzxi-0 iNYXqo"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              Create Account
            </span>
          </button>
        </div>
        <div
          class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-wrap-wrap ant-flex-justify-center ant-flex-gap-large"
          style="margin-bottom: 20px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomePage2 matches the snapshot mobile 1`] = `
<div>
  <div
    class="FourImagesBgstyles__DivImageContainer-sc-1fpm6f5-0 ckwQRQ"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-center"
      style="height: 100vh;"
    >
      <div
        class="ant-layout css-dev-only-do-not-override-1uq9j6g"
        style="width: 100%; max-width: 300px; padding: 20px 15px;"
      >
        <h1
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
          style="text-align: center;"
        >
          Welcome
        </h1>
        <input
          class="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          placeholder="email"
          style="margin-bottom: 20px;"
          type="text"
          value=""
        />
        <div
          class="Welcome__ButtonContainer-sc-e8hzxi-0 iNYXqo"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              email
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              LogIn
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-block"
            type="button"
          >
            <span>
              Create Account
            </span>
          </button>
        </div>
        <div
          class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-wrap-wrap ant-flex-justify-center ant-flex-gap-large"
          style="margin-bottom: 20px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              G
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-circle ant-btn-primary"
            type="button"
          >
            <span>
              VK
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
